pkg/kubelet/kubelet.go

```diff
@@ -145,9 +145,12 @@ const (
 	// DefaultContainerLogsDir is the location of container logs.
 	DefaultContainerLogsDir = "/var/log/containers"
 
-	// MaxContainerBackOff is the max backoff period, exported for the e2e test
+	// MaxContainerBackOff is the max backoff period for container restarts, exported for the e2e test
 	MaxContainerBackOff = 300 * time.Second
 
+	// MaxImageBackOff is the max backoff period for image pulls, exported for the e2e test
```

Comment:
In this is actually used by the e2e tests, I'd expect some e2e tests to migrate to this value in this PR. Have you audited all the uses of `MaxContainerBackoff`?

Reply:
btw, if we don't have any - with CRI Proxy you can now test many different scenarios. See https://github.com/kubernetes/kubernetes/pull/121604/ for example use

Reply:
to be honest this comment is disingenuous, at least in the context of this (mini) PR, because there are no e2e tests for the image pull behavior, just the container restart behavior. that being said the linked pr looks cool 🚀  and relevant so I am rebasing on that

Reply:
And to more directly answer your question, the usages of `MaxContainerBackoff` are all in /home/lauralorenz_google_com/go/src/k8s.io/kubernetes/test/e2e/common/node/pods.go, specifically the conformance test steps [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/common/node/pods.go#L677-L771), which do not test the imagepullbackoff, but the container restart (somewhat confusingly -- testing behavior /triggered/ by an image pull -- but its not testing the individual backoff caused by ImagePullBackOff errors, but the container's backoff)

Reply:
yes, the test you are pointed to only test container backoff. Mostly because it was easy before. This is why we introduced the cri proxy.

In order to get this KEP to alpha we need to have a test that validates that the image pull back off is not affected by the change you apply by configuring containers backoff. introducing this test here will simplify testing the actual configuration feature

Reply:
FYI, I'm working on such a test in an derivative PR https://github.com/kubernetes/kubernetes/pull/128559, but having trouble running the e2e node tests locally so borrowing the prow infra there 😈